shell/probe/demo.ps1

```diff
@@ -37,19 +37,39 @@ function DownloadTest {
 }
 
 function ExecuteTest {
-    & $TempFile
-    if ($LASTEXITCODE -eq 100 ) {
+    try {
+        & $TempFile
```

Comment:
Use the following so that you can get the exit staus correctly
```
$p = Start-Process -FilePath <path> -ArgumentList <args> -Wait -NoNewWindow -PassThru
$p.ExitCode
```

Reply:
Does that provide different results for you? It doesn't for me

Reply:
It may or may not, depending on the happy or unhappy paths. on the latter, you will see differences. 
but regardless, we need start process to run tests in a child process.

Reply:
done